kernel/kernel-api/src/main/java/io/delta/kernel/internal/skipping/StatsSchemaHelper.java

```diff
@@ -0,0 +1,258 @@
+package io.delta.kernel.internal.skipping;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import io.delta.kernel.expressions.Column;
+import io.delta.kernel.expressions.Literal;
+import io.delta.kernel.internal.util.Tuple2;
+import io.delta.kernel.types.*;
+import static io.delta.kernel.internal.util.ColumnMapping.getPhysicalName;
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+/**
+ * Provides information and utilities for statistics columns given a table schema. Specifically,
+ * it is used to:
+ * <ol>
+ *     <li>Get the expected statistics schema given a table schema</li>
+ *     <li>Check if a {@link Literal} or {@link Column} is skipping eligible</li>
+ *     <li>Get the statistics column for a given stat type and logical column</li>
+ * </ol>
+ */
+public class StatsSchemaHelper {
+
+    //////////////////////////////////////////////////////////////////////////////////
+    // Private static fields and methods
+    //////////////////////////////////////////////////////////////////////////////////
+
+    private static String NUM_RECORDS = "numRecords";
+    private static String MIN = "minValues";
+    private static String MAX = "maxValues";
+    private static String NULL_COUNT = "nullCount";
+
+    private static Set<String> skippingEligibleTypeNames = new HashSet<String>() {
+        {
+            add("byte");
+            add("short");
+            add("integer");
+            add("long");
+            add("float");
+            add("double");
+            add("date");
+            add("timestamp");
+            add("string");
+        }
+    };
+
+    /**
+     * Returns true if the given data type is eligible for MIN/MAX data skipping.
+     */
+    private static boolean isSkippingEligibleDataType(DataType dataType) {
+        return skippingEligibleTypeNames.contains(dataType.toString()) ||
+            dataType instanceof DecimalType;
+    }
+
+    /**
+     * Given a data schema returns the expected schema for a min or max statistics column. This
+     * means 1) replace logical names with physical names 2) set nullable=true 3) only keep stats
+     * eligible fields (i.e. don't include fields with isSkippingEligibleDataType=false)
```

Comment:
for this and the following method can you put the structure of the stats in some way? Maybe a pretty json. So that we can validate whether the code matches that. 

Also then this will become a great reference material for future.

Reply:
and do these need to be public methods? i dont see it being called from anywhere outside this class

Reply:
Not sure which public method you're referring to?

Reply:
I added an example of a data schema and stats to the top level docs.